annlite/executor.py

```diff
@@ -9,57 +9,47 @@
 from jina import Executor, requests
 from jina.logging.logger import JinaLogger
 
-from .index import AnnLite
-
 
 class AnnLiteIndexer(Executor):
     """
-    A simple indexer that wraps the AnnLite indexer and adds a simple interface for indexing and searching.
+    :param n_dim: Dimensionality of vectors to index
+    :param dim: Deprecated, use n_dim instead
+    :param metric: Distance metric type. Can be 'euclidean', 'inner_product', or 'cosine'
+    :param include_metadata: If True, return the document metadata in response
+    :param limit: Number of results to get for each query document in search
+    :param ef_construction: The construction time/accuracy trade-off
+    :param ef_search: The query time accuracy/speed trade-off
+    :param max_connection: The maximum number of outgoing connections in the
+        graph (the "M" parameter)
+    :param index_access_paths: Default traversal paths on docs
+            (used for indexing, delete and update), e.g. '@r', '@c', '@r,c'
+    :param search_access_paths: Default traversal paths on docs
+    (used for search), e.g. '@r', '@c', '@r,c'
+    :param columns: A list or dict of column names to index.
     """
 
     def __init__(
         self,
         n_dim: int = 0,
         metric: str = 'cosine',
         limit: int = 10,
-        ef_construction: int = 200,
-        ef_query: int = 50,
-        max_connection: int = 16,
+        data_path: Optional[str] = None,
```

Comment:
if None, docarray will generate random name each time. I'd say let's keep one default value, so that users who do not specify this will see persisted data

Reply:
also let's document this parameter

Reply:
`data_path` will be given a specific value below in `config` dict if it keeps `None`.

Reply:
A bug there, I will fix it to this:
```python
config={
    'n_dim`: n_dim,
    'metric`: metric,
    'ef_construction`: ef_construction,
    'ef_search': ef_search,
    'max_connection': max_connection,
    'data_path': data_path or self.workspace or './workspace',
    'columns': columns,
}
```

Reply:
In the context of executor, actually `data_path` is not recommended, and would not work with multi-shards (different shards writes to the same path `data_path`).

Reply:
@numb3r3 Then where to store the indexed data?

Reply:
I would suggest to keep it like this at this moment.